docs/README.md

```diff
@@ -0,0 +1,139 @@
+
+Here's the [FAQ](./faq.md)
+
+Nix-Doom-Emacs uses [nix-straight.el](https://github.com/nix-community/nix-straight.el) under the hood to install dependencies. It's a low level wrapper to add Nix integration over straight.el, the declarative package manager used by Doom Emacs. 
+
+If you're not aware yet, then:
+
+#### **WARNING**: HERE BE DRAGONS! THIS IS A FRAGILE PROJECT
+
+# Getting Started
+
+To get started, we suggest these methods. They are ordered from most suggested to least suggested.
+In all of these methods, you'll need your Doom Emacs configuration. It should contain the following three files: 
+`config.el`, `init.el` and `packages.el`. If you don't already have an existing `doom-emacs` configuration, you can use the contents of `test/doom.d` as a template.
+
+The Doom configuration will be referred to as `./doom.d` in these snippets. You can name it whatever you like.
+
+## Flake + Home-Manager
+
+`File: flake.nix`
+```nix
+{
+  inputs = {
+    home-manager.url = "github:nix-community/home-manager";
+    nix-doom-emacs.url = "github:nix-community/nix-doom-emacs";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    lib,
+    home-manager,
+    nix-doom-emacs,
+    ...
+  }:
+    let
+      system      = "x86_64-linux";
+      specialArgs = { inherit inputs; };
+    in {
+    nixosConfigurations.exampleHost = lib.nixosSystem {
+      inherit system specialArgs;
+      modules = [
+        ./default.nix
+        home-manager.nixosModules.home-manager
+        {
+          home-manager.users.exampleUser.imports = [ ./home.nix ];
+        }
+      ];
+    };
+  };
+}
+
+```
+
+`File: home.nix`
+```nix
+{ config, pkgs, inputs, ... }: {
+  imports = [ inputs.nix-doom-emacs.hmModule ];
+
+  # ...
+  programs.doom-emacs = {
+    enable = true;
+    doomPrivateDir = ./doom.d; # Directory containing your config.el, init.el
+                               # and packages.el files
+  };
+  # ...
+}
+```
+
+## Non-Flake Home-Manager
+
+```nix
+{ pkgs, ... }:
+
+let
+  doom-emacs = pkgs.callPackage (builtins.fetchTarball {
+    url = https://github.com/nix-community/nix-doom-emacs/archive/master.tar.gz;
+  }) {
+    doomPrivateDir = ./doom.d;  # Directory containing your config.el, init.el
+                                # and packages.el files
+  };
+in {
+  home.packages = [ doom-emacs ];
+}
+```
+
+
+## Standalone
```

Comment:
```suggestion
## NixOS
```

I think we also need a "standalone" section, but this would give as an example a `shell.nix` file.

Reply:
I'm not quite sure how a `flake.nix` file with a `devShell` for standalone systems would look like, would you provide an example snippet?

Reply:
For `shell.nix`:

```
{ pkgs ? import <nixpkgs> { } }:

let
  repo = pkgs.fetchFromGitHub {
    owner = "nix-community";
    repo = "nix-doom-emacs";
    rev = "<commit>";
    sha256 = "<hash>";
  };
  nix-doom-emacs = pkgs.callPackage (import repo) {
    doomPrivateDir = ./doom.d;
  };
in
pkgs.mkShell {
  buildInputs = [ nix-doom-emacs ];
}
```

Reply:
For `flake.nix`:

```nix
{
  description = "nix-doom-emacs shell";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
    nix-doom-emacs.url = "github:nix-community/nix-doom-emacs";
  };

  outputs = { self, nixpkgs, nix-doom-emacs, ... }:
  let
    system = "x86_64-linux";
    pkgs = import nixpkgs { inherit system; };
    doom-emacs = nix-doom-emacs.packages.${system}.default.override {
      doomPrivateDir = ./doom.d;
    };
  in
  {
    devShells.${system}.default = pkgs.mkShell {
      buildInputs = [ doom-emacs ];
    };
  };
}
```